cgo/pow/pow.go

```diff
@@ -0,0 +1,78 @@
+package pow
```

Comment:
Should the filename be revised to `cgo/pow/pow_dcurl.go`?

Reply:
Originally, I use `pow_dcurl.go`. 
Under the module name dcurl.go or `github.com/DLTcollab/dcurl/cgo` now, 
it should be clear that pow here is using dcurl.
Hence I thought the postfix could be redundant.

Reply:
In `iota.go/pow`, they have different postfix for different pow implementations.